certif/app/services/session.js

```diff
@@ -19,4 +31,32 @@ export default class CurrentSessionService extends SessionService {
   handleInvalidation() {
     super.handleInvalidation('/connexion');
   }
+
+  async handleLocale({ isFranceDomain, localeFromQueryParam, userLocale }) {
+    this._localeFromQueryParam = localeFromQueryParam;
+
+    if (isFranceDomain) {
+      this._setLocale(FRENCH_INTERNATIONAL_LOCALE);
+
+      if (!this.locale.hasLocaleCookie()) {
+        this.locale.setLocaleCookie(FRENCH_FRANCE_LOCALE);
+      }
+
+      return;
+    }
+
+    if (this._localeFromQueryParam) {
+      this._setLocale(this._localeFromQueryParam);
+      return;
+    }
```

Comment:
Pourquoi pas directement: 
```suggestion
    if (localeFromQueryParam) {
      this._setLocale(localeFromQueryParam);
      return;
    }
```

Reply:
`this._localeFromQueryParam` permet de garder en mémoire la locale utilisée pour charger l'application la première fois. Cela correspond au cas de test suivant qui fait que le query param utilisé pour chargé l'application va être « gardé en mémoire » même si l'utilisateur s'authentifie par la suite : 
> 7. Aller sur Pix Certif .org en faisant varier les valeurs du paramètre lang, en mettant alternativement ?lang=fr https://certif-pr5948.review.pix.org/connexion?lang=fr et ?lang=en https://certif-pr5948.review.pix.org/connexion?lang=en et constater que le query param est bien pris en compte

Reply:
Et désolé cela ne correspond pas à un cas de test fonctionnel décrit dans la PR (comme j'ai pu le mettre dans un bout de commentaire où je mentionnais le cas 7. et que je viens de supprimer à l'instant).

Reply:
@aceol pour démontrer que l'utilisation de `this._localeFromQueryParam` est nécessaire, j'ai ajouté un test d'acceptance qui plante si cette propriété n'est pas présente ou si on utilise directement `localeFromQueryParam` plutôt que `this._localeFromQueryParam`. Ce test d'acceptance est nécessaire car les tests unitaires ne testent pas ce cas du chargement de l'application **puis** de l'authentification de l'utilisateur.

Et cela a mis en lumière le manque de la condition ci-dessous, que je viens d'ajouter, pour « garder en mémoire » la `localeFromQueryParam` et « ne pas l'oublier » lors d'un appel à `handleLocale` sans `localeFromQueryParam` :

```javascript
async handleLocale({ isFranceDomain, localeFromQueryParam, userLocale }) {
    if (localeFromQueryParam) {
      this._localeFromQueryParam = localeFromQueryParam;
    }
  // ...
```

Voilà c'était la dernière modification sur cette PR. Merci à toutes et à tous pour vos revues !